transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/insight/SelectTransferMethodInsightTest.java

```diff
@@ -0,0 +1,159 @@
+import com.hyperwallet.android.ui.testutils.rule.HyperwalletExternalResourceManager;
+import com.hyperwallet.android.ui.testutils.rule.HyperwalletMockWebServer;
+import com.hyperwallet.android.ui.testutils.rule.HyperwalletSdkRule;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepositoryFactory;
+import com.hyperwallet.android.ui.transfermethod.rule.HyperwalletInsightMockRule;
+import com.hyperwallet.android.ui.transfermethod.view.SelectTransferMethodActivity;
+import com.hyperwallet.android.ui.user.repository.UserRepositoryFactory;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import java.util.Map;
+
+public class SelectTransferMethodInsightTest {
+    @ClassRule
+    public static HyperwalletExternalResourceManager sResourceManager = new HyperwalletExternalResourceManager();
+
+    @Rule
+    public HyperwalletSdkRule mHyperwalletSdkRule = new HyperwalletSdkRule();
+    @Rule
+    public HyperwalletInsightMockRule mHyperwalletInsightMockRule = new HyperwalletInsightMockRule();
+    @Rule
+    public HyperwalletMockWebServer mMockWebServer = new HyperwalletMockWebServer(8080);
+    @Rule
+    public ActivityTestRule<SelectTransferMethodActivity> mActivityTestRule =
+            new ActivityTestRule<>(SelectTransferMethodActivity.class, true, false);
+    @Rule
+    public MockitoRule mMockito = MockitoJUnit.rule();
+    @Captor
+    private ArgumentCaptor<Map<String, String>> mParamsCaptor;
+
+    @Before
+    public void setup() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("authentication_token_response.json")).mock();
+        IdlingRegistry.getInstance().register(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @After
+    public void cleanup() {
+        TransferMethodRepositoryFactory.clearInstance();
+        UserRepositoryFactory.clearInstance();
+        IdlingRegistry.getInstance().unregister(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @Test
+    public void testSelectTransferMethod_verifyInsightEventCreatedOnTransferMethodOptionsLoad() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("user_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_keys_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                timeout(5000).times(1)).trackImpression(any(Context.class),
+                eq(HyperwalletInsight.PAGE_TRANSFER_METHOD_SELECT),
```

Comment:
FYI,
I think here we need to check with hard coded expected raw string value instead of referencing the actual defined variable from development block. Because if developer made typo mistake, test case cannot identify what issue happened. I can see this technique is implemented many places like ListPrepaidCardReceiptsTest.java(which is pushed to same PR). I can see XML string resource reference also.

Reply:
Thank you for the suggestion, I thought so too at first but we have agreed to use reference strings instead of hard coded values due to the fact that we may have to support multiple languages in the future.